cluster/src/main/java/io/scalecube/cluster/Cluster.java

```diff
@@ -95,22 +96,18 @@ static Cluster joinAwait(ClusterConfig config) {
   /**
    * Init cluster instance with the given configuration and join cluster synchronously.
    */
-  static CompletableFuture<Cluster> join(final ClusterConfig config) {
-    return new ClusterImpl(config).join0();
+  static Mono<Cluster> join(final ClusterConfig config) {

```

Comment:
Plz dont substitute `futures` to reactor `publishers`. Let's do this in separate PR.

Reply:
i guess if we use rsocket it comes more native to use reactor rather then futures